### ISSUE TITLE
Segfault when serializing svg with xlink:title attribute

### ISSUE THREAD

**Issue by noam3127** [lexbor/lexbor#228]
First of all, thank you for this amazing library!

When parsing + serializing the following html using `lxb_html_serialize_tree_str()`, a segfault occurs:

```html
<!DOCTYPE html>
<html>

<body>
  <svg xlink:title="arrow">
  </svg>
</body>

</html>
```

This issue happens only when it's a `<svg>` element that contains a `xlink:title` attribute. As far as I can tell, no other `xlink` namespaced attribute has this problem.

Thank you for your time.

**Comment by lexborisov** [comment]
Hi @noam3127 

What version of the library are you using? I tried it on the current code and everything works fine.

```C
    lxb_status_t status;
    lexbor_str_t str;
    lxb_html_document_t *document;

    static const lxb_char_t html[] =
	"<!DOCTYPE html><html><body><svg xlink:title=\"arrow\"></svg></body></html>";

    document = lxb_html_document_create();
    status = lxb_html_document_parse(document, html, sizeof(html) - 1);
    if (status != LXB_STATUS_OK) {
        return EXIT_FAILURE;
    }

    str.data = NULL;

    lxb_html_serialize_tree_str(lxb_dom_interface_node(document), &str);
```

Result:
```
<!DOCTYPE html><html><head></head><body><svg xlink:title="arrow"></svg></body></html>
```

**Comment by noam3127** [comment]
We're using v2.1.0. Is this something that might have gotten fixed in recent versions?

**Comment by lexborisov** [comment]
@noam3127 

Try using a newer version, such as v2.3.0, or the latest one from the repository.
Please let me know if the problem has gone away.

**Comment by noam3127** [comment]
Ok, after building from the latest source and testing locally, the problem is gone (as you predicted). 
FWIW, this commit - https://github.com/lexbor/lexbor/commit/7ecf3661e000e504d6b7406a430a7b965a83fbc2 is the code that apparently fixed this bug, as testing any build from before then leads to that segfault. 
Thanks again for your help.

**Comment by lexborisov** [comment]
@noam3127 

That's an interesting side effect.  I'm thinking of parsing what exactly was there and why this patch with the addition of cloning fixed the problem.
At the very least to make sure there's nothing wrong there.

Thank you for your active participation!

**Comment by nielsdos** [comment]
I also performed a bisect and found a different commit that fixed the issue: 842754c0863c0d8d03a6ab8bb449f631139d99d5

This makes more sense. Prior to this commit the following happened:

The segfault happens because the address 0xba is dereferenced in `lxb_html_serialize_attribute_cb` when accessing `lexbor_hash_entry_str(&data->entry)`. This is because `data` is equal to 0xba. Notice that `LXB_TAG_TITLE` is equal to 0xba and we're dealing with a (xlink:)title tag. Given the fix commit this indeed makes sense.

**Comment by lexborisov** [comment]
@noam3127  @nielsdos 

thank you for all your hard work! I think I'll watch it tomorrow as well. Unfortunately, due to busyness I forgot about it.

**Comment by lexborisov** [comment]
I looked into this issue and yes the problem was fixed by patch [842754c](https://github.com/lexbor/lexbor/commit/842754c0863c0d8d03a6ab8bb449f631139d99d5). Tests for this are present in the patch.
